vividus-extension-selenium/src/main/java/org/vividus/monitor/GenericHarOnFailureMonitor.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2019-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.monitor;
+
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import com.browserup.harreader.model.Har;
+import com.browserup.harreader.model.HarEntry;
+import com.google.common.eventbus.EventBus;
+
+import org.vividus.context.RunContext;
+import org.vividus.proxy.IProxy;
+import org.vividus.selenium.IWebDriverProvider;
+import org.vividus.util.json.JsonUtils;
+
+public class GenericHarOnFailureMonitor extends AbstractPublishingAttachmentOnFailureMonitor
+{
+    private static final String NO_HAR_ON_FAILURE_META_NAME = "noHarOnFailure";
+
+    private boolean harOnFailure;
+
+    private IProxy proxy;
+    private final List<HarEntry> harEntries = Collections.synchronizedList(new ArrayList<>());
```

Comment:
it won't allow to make the monitor thread-safe: all threads will use single collection -> as result it will contain ALL har entries for ALL threads, there is going to be a chaos